extensions/http-curl/tests/HTTPHandlers.h

```diff
@@ -792,20 +792,40 @@ class C2AcknowledgeHandler : public ServerAwareHandler {
     rapidjson::Document root;
     root.Parse(req.data(), req.size());
     if (root.IsObject() && root.HasMember("operationId")) {
-      std::lock_guard<std::mutex> guard(mtx_);
+      std::lock_guard<std::mutex> guard(ack_operations_mtx_);
```

Comment:
With locking two mutexes, you always have to be careful. Is it possible to have another thread locking them in the reverse order than here? So, first `apply_count_mtx_` and then `ack_operations_mtx_`? That could result in a deadlock. It's possible by calling `isAcknowledged` first and `getApplyCount` second, for example.

Reply:
`ack_operations_mtx_` is released before locking `apply_count_mtx_`, so they are never locked at the same time. Since only one is held at a time, it's not possible that this function and another thread are waiting for each other.